statsprinter.go

```diff
@@ -26,7 +36,46 @@ var (
 	colorLightCyan   = color.LightCyan.Printf
 )
 
-type planePrinter struct{}
+type planePrinter struct {
+	showTimestamp *bool
+}
+
+type PrintOptions struct {
+	color         Color
+	message       string
+	showTimestamp *bool
+}
+
+func printReply(options *PrintOptions) {
+	if options.showTimestamp != nil && *options.showTimestamp {
+		t := time.Now().Format(timeFormat)
+		options.message = fmt.Sprintf("%s %s", t, options.message)
+	}
+	switch options.color {
```

Comment:
Only 2 functions use printReply now, So we should be okay to remove most of these switchcases and leave something like Yellow to be the default color. What do you think @pouriyajamshidi ?

Reply:
hmmmm, right.

Or you could also just pass the color as an argument to it instead of using enums

Reply:
You are not wrong. But we are looking at the problem from two different perspectives.

What I mean to say is, we do not need all of this "fancy" stuff to implement this feature.

If we only add an extra check for the `-D` flag in `printProbeSuccess` and `printProbeFail`, we then do not need enums, `printReply` function and the `PrintOptions` struct. 

A quick and dirty (yet working) implementation for `printProbeSuccess` is:

```go
func (p *planePrinter) printProbeSuccess(hostname, ip string, port uint16, streak uint, rtt float32) {
	if hostname == "" && *p.showTimestamp == false {
		colorLightGreen("Reply from %s on port %d TCP_conn=%d time=%.3f ms\n",
			ip, port, streak, rtt)
	} else if hostname == "" && *p.showTimestamp {
		t := time.Now().Format(timeFormat)
		colorLightGreen("%s Reply from %s on port %d TCP_conn=%d time=%.3f ms\n",
			t, ip, port, streak, rtt)
	} else if hostname != "" && *p.showTimestamp == false {
		colorLightGreen("Reply from %s (%s) on port %d TCP_conn=%d time=%.3f ms\n",
			hostname, ip, port, streak, rtt)
	} else {
		t := time.Now().Format(timeFormat)
		colorLightGreen("%s Reply from %s (%s) on port %d TCP_conn=%d time=%.3f ms\n",
			t, hostname, ip, port, streak, rtt)
	}
}
```

Reply:
okay I understand, I've tweaked my code to follow a pattern similar to this